scripts/dev/newReleaseBranch.sh

```diff
@@ -0,0 +1,680 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOT_DIR="${SCRIPT_DIR}/../../.."
+
+usage() {
+    cat << ???  1>&2;
+
+Creates HotFix and ServicePack (.N) branches for repo, share and packaging projects, and updates the master
+or ServicePack branches ready for the next release.
+
+Usage: $0 <hotFixVersion> [-v <masterVersion>] [-h] [-l]
+  <hotFixVersion>:  HotFix Branch to be created or modified.
+  -v <masterVersion>: Overrides the next development version on master so that the HotFix major version may be changed.
+                    Ignored if the master branch is not going to be modified. Must contain 3 integers separated by dots.
+  -h: Display this help
+  -l: Output extra logging
+  -t: use test branches: release/test/master and release/test/X.x.x
+  -c: cleanup (delete) local test release/test/X.x.x branches that are about to be created or modified.
+  -s: skip the push to the remote git repository
+
+Examples:
+  1. After the release of 23.1.0 from the master branch:
+     $ $0 23.1.0
+     Creates the HotFix branch for 23.1.0, creates the 23.1.N ServicePack branch and modifies the master branch for
+     use by the next minor version 23.2.0.
+
+  2. After the release of 23.1.1 from the ServicePack branch release/23.1.N:
+     $ $0 23.1.1
+     Creates the HotFix branch for 23.1.1 and modifies the 23.1.N ServicePack branch for use by the next version 23.1.2
+
+  3. After the release of 23.1.2 from the ServicePack branch release/23.1.N
+     $ $0 23.1.2
+     Creates the HotFix branch for 23.1.2 and modifies the 23.1.N ServicePack branch for use by the next version 23.1.3
```

Comment:
I think this example is basically the same as the previous.

Reply:
Yes it is. The examples are linked. We generally do a major or minor release followed by TWO service pack releases and the next major or minor release. The for examples show this:
- 23.1.0
- 23.1.1
- 23.1.2   <-- the second SP that you correctly identified as basically the same as the previous one.
- 25.2.0   <-- under the year based versions 23 jumps to 25